src/ImageSharp/Formats/SpecializedImageDecoder{T}.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Formats;
+
+/// <summary>
+/// Acts as a base class for specialized image decoders.
+/// Specialized decoders allow for additional options to be passed to the decoder.
+/// Types that inherit this decoder are required to implement cancellable synchronous decoding operations only.
+/// </summary>
+/// <typeparam name="T">The type of specialized options.</typeparam>
+public abstract class SpecializedImageDecoder<T> : ImageDecoder, ISpecializedImageDecoder<T>
+    where T : ISpecializedDecoderOptions
+{
+    /// <inheritdoc/>
+    public Image<TPixel> Decode<TPixel>(T options, Stream stream)
+        where TPixel : unmanaged, IPixel<TPixel>
+        => WithSeekableStream(
+            options.GeneralOptions,
+            stream,
+            s => this.Decode<TPixel>(options, s, default));
+
+    /// <inheritdoc/>
+    public Image Decode(T options, Stream stream)
+        => WithSeekableStream(
+            options.GeneralOptions,
+            stream,
+            s => this.Decode(options, s, default));
+
+    /// <inheritdoc/>
+    public Task<Image<TPixel>> DecodeAsync<TPixel>(T options, Stream stream, CancellationToken cancellationToken)
+        where TPixel : unmanaged, IPixel<TPixel>
+        => WithSeekableMemoryStreamAsync(
+            options.GeneralOptions,
+            stream,
+            (s, ct) => this.Decode<TPixel>(options, s, ct),
+            cancellationToken);
+
+    /// <inheritdoc/>
+    public Task<Image> DecodeAsync(T options, Stream stream, CancellationToken cancellationToken)
+        => WithSeekableMemoryStreamAsync(
+            options.GeneralOptions,
+            stream,
+            (s, ct) => this.Decode(options, s, ct),
+            cancellationToken);
+
+    /// <summary>
+    /// Decodes the image from the specified stream to an <see cref="Image{TPixel}" /> of a specific pixel type.
+    /// </summary>
+    /// <remarks>
+    /// This method is designed to support the ImageSharp internal infrastructure and is not recommended for direct use.
+    /// </remarks>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    /// <param name="options">The specialized decoder options.</param>
+    /// <param name="stream">The <see cref="Stream" /> containing image data.</param>
+    /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+    /// <returns>The <see cref="Image{TPixel}" />.</returns>
+    /// <exception cref="ImageFormatException">Thrown if the encoded image contains errors.</exception>
+    protected abstract Image<TPixel> Decode<TPixel>(T options, Stream stream, CancellationToken cancellationToken)
```

Comment:
Here we still lack the option for defaulting `CancellationToken`.

Reply:
That happens on the public caller. This is the protected inner call.

Reply:
Oops, commented on the wrong method.

If you type `new JpegDecoder().DecodeAsync( ...`  you won't see an overload that takes `JpegDecoderOptions` without also asking for a `CancellationToken`.

One reason why I aim to add tests that validate all public methods directly it helps avoiding such mistakes.

Reply:
Ah right... That was intentional. My thought was that more often than not a cancellation token will be passed. I'll update.

Reply:
Fixed. I've updated all tests calling those APIs directly to remove the default passed token.